Source/SwiftLintBuiltInRules/Rules/Metrics/NestingRule.swift

```diff
@@ -12,137 +13,171 @@ struct NestingRule: Rule {
-        let isTypeOrExtension = SwiftDeclarationKind.typeKinds.contains(declarationKind)
-            || SwiftDeclarationKind.extensionKinds.contains(declarationKind)
-        let isFunction = SwiftDeclarationKind.functionKinds.contains(declarationKind)
+        override func visit(_ node: ActorDeclSyntax) -> SyntaxVisitorContinueKind {
+            validate(node, triggeringToken: node.actorKeyword, inFunction: validationData.isFunction)
+            return .visitChildren
+        }
 
-        guard isTypeOrExtension || isFunction else {
-            return validate(file: file, substructure: dictionary.substructure,
-                            args: args.with(previousKind: structureKind))
+        override func visitPost(_ node: ActorDeclSyntax) {
+            validationData.pop()
         }
 
-        let currentTypeLevel = isTypeOrExtension ? args.typeLevel + 1 : args.typeLevel
-        let currentFunctionLevel = isFunction ? args.functionLevel + 1 : args.functionLevel
+        override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+            validate(node, triggeringToken: node.classKeyword, inFunction: validationData.isFunction)
+            return .visitChildren
+        }
 
-        var violations = args.violations
+        override func visitPost(_ node: ClassDeclSyntax) {
+            validationData.pop()
+        }
 
-        if let violation = levelViolation(file: file, dictionary: dictionary,
-                                          previousKind: args.previousKind,
-                                          level: isFunction ? currentFunctionLevel : currentTypeLevel,
-                                          forFunction: isFunction) {
-            violations.append(violation)
+        override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+            validate(node, triggeringToken: node.enumKeyword, inFunction: validationData.isFunction)
+            return .visitChildren
         }
 
-        return validate(file: file, substructure: dictionary.substructure,
-                        args: ValidationArgs(
-                            typeLevel: currentTypeLevel,
-                            functionLevel: currentFunctionLevel,
-                            previousKind: structureKind,
-                            violations: violations
-            )
-        )
-    }
+        override func visitPost(_ node: EnumDeclSyntax) {
+            validationData.pop()
+        }
 
-    private func levelViolation(file: SwiftLintFile, dictionary: SourceKittenDictionary,
-                                previousKind: SwiftStructureKind?, level: Int, forFunction: Bool) -> StyleViolation? {
-        guard let offset = dictionary.offset else {
-            return nil
+        override func visit(_ node: ExtensionDeclSyntax) -> SyntaxVisitorContinueKind {
+            validate(node, inFunction: validationData.isFunction)
```

Comment:
Even though the rule cannot trigger on extensions, you may pass the `extensionKeyword` to the `validate` method to avoid the nilable argument.